cmd/writePipelineEnv.go

```diff
@@ -31,13 +31,21 @@ func WritePipelineEnv() *cobra.Command {
 }
 
 func runWritePipelineEnv() error {
-	inBytes, err := ioutil.ReadAll(os.Stdin)
-	if err != nil {
-		return err
+	pipelineEnv, ok := os.LookupEnv("PIPER_pipelineEnv")
```

Comment:
Do you need to unset the variable once read? How about parallel executions?

Reply:
OS Variables don't need to be unset and are not shared globally. They are set in a parent process and are passed to child processes. So in parallel execution every process has their own `PIPER_pipelineEnv`

Reply:
Also unsetting a environment variable has therefore no effect to parent processes. Just for processes that would be executed by the go code itself